### PR TITLE
[SPARK-46535][SQL] Fix NPE when describe extended a column without col stats

### PR DESCRIPTION
**Pull request by Zouxxyy**
### Why are the changes needed?

Currently executing DESCRIBE TABLE EXTENDED a column without col stats with v2 table will throw a null pointer exception.

```text
Cannot invoke "org.apache.spark.sql.connector.read.colstats.ColumnStatistics.min()" because the return value of "scala.Option.get()" is null
java.lang.NullPointerException: Cannot invoke "org.apache.spark.sql.connector.read.colstats.ColumnStatistics.min()" because the return value of "scala.Option.get()" is null
	at org.apache.spark.sql.execution.datasources.v2.DescribeColumnExec.run(DescribeColumnExec.scala:63)
	at org.apache.spark.sql.execution.datasources.v2.V2CommandExec.result$lzycompute(V2CommandExec.scala:43)
	at org.apache.spark.sql.execution.datasources.v2.V2CommandExec.result(V2CommandExec.scala:43)
	at org.apache.spark.sql.execution.datasources.v2.V2CommandExec.executeCollect(V2CommandExec.scala:49)
	at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.$anonfun$applyOrElse$1(QueryExecution.scala:118)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId0$6(SQLExecution.scala:150)
	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:241)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId0$1(SQLExecution.scala:116)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:918)
```

This RP will fix it

### How was this patch tested?

Add a new test `describe extended (formatted) a column without col stats`

### CODE REVIEW

**Comment by Zouxxyy** [comment]
@huaxingao Can you help with a review~

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/DescribeTableSuite.scala [hunk 1]
```diff
@@ -175,4 +175,25 @@ class DescribeTableSuite extends command.DescribeTableSuiteBase
           Row("max_col_len", "NULL")))
     }
   }
+
+  test("describe extended (formatted) a column without col stats") {
```
Comment (yaooqinn) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/DescribeTableSuite.scala:179]:
```suggestion
  test("SPARK-39859: describe extended (formatted) a column without col stats") {
```

**Review by yaooqinn — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/DescribeTableSuite.scala [hunk 2]
```diff
@@ -176,7 +176,7 @@ class DescribeTableSuite extends command.DescribeTableSuiteBase
     }
   }
 
-  test("describe extended (formatted) a column without col stats") {
+  test("SPARK-39859: describe extended (formatted) a column without col stats") {
```
Comment (Zouxxyy) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/DescribeTableSuite.scala:179]: @yaooqinn I'm not using the new JIRA id, is it appropriate to be added here too?

Comment (yaooqinn) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/DescribeTableSuite.scala:179]: Please create a new JIRA ticket to track, SPARK-39859 is not suitable here or in the title.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/DescribeTableSuite.scala [hunk 3]
```diff
@@ -175,4 +175,25 @@ class DescribeTableSuite extends command.DescribeTableSuiteBase
           Row("max_col_len", "NULL")))
     }
   }
+
+  test("SPARK-39859: describe extended (formatted) a column without col stats") {
```
Comment (yaooqinn) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/DescribeTableSuite.scala:179]:
```suggestion
  test("SPARK-46535: describe extended (formatted) a column without col stats") {
```

**Review by yaooqinn — APPROVED** [review]

**Comment by MaxGekk** [comment]
Not related to this PR but users shouldn't see NPE. We should convert NPE to an internal error. cc @cloud-fan

**Review by MaxGekk — APPROVED** [review]

**Review by LuciferYang — APPROVED** [review]

**Comment by MaxGekk** [comment]
@Zouxxyy This should be ported to only `branch-3.5`? The earlier version (`branch-3.4`) doesn't have the issue according to https://issues.apache.org/jira/browse/SPARK-46535, correct?

**Comment by Zouxxyy** [comment]
> @Zouxxyy This should be ported to only `branch-3.5`? The earlier version (`branch-3.4`) doesn't have the issue according to https://issues.apache.org/jira/browse/SPARK-46535, correct?

yes

**Comment by MaxGekk** [comment]
The failed tests are not related to the changes.

Merging to master/3.5. Thank you, @Zouxxyy and @yaooqinn @LuciferYang for review.

**Comment by MaxGekk** [comment]
@Zouxxyy Congratulations with your first contribution to Apache Spark!

**Pull request closed without merging**